### PR TITLE
Expose flags for max-pods-calculator to use a configurable max pods ceiling

### PR DESCRIPTION
**Pull request by a7i**
*Issue #, if available:*

*Description of changes:*
Provide the ability to pass in the ceiling for max-pods instead of relying on the default values for low/high cpu count.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by a7i** [comment]
@overdrive3000 this is a pretty small change, can this be reviewed?

File: files/max-pods-calculator.sh [hunk 1]
```diff
@@ -140,12 +147,16 @@ else
     max_pods=$(calculate_max_ip_addresses_secondary_ips $enis_for_pods $INSTANCE_MAX_ENIS_IPS)
 fi
 
-# Limit the total number of pods that can be launched on any instance type based on the vCPUs on that instance type.
-MAX_POD_CEILING_FOR_LOW_CPU=110
-MAX_POD_CEILING_FOR_HIGH_CPU=250
-CPU_COUNT=$(echo $DESCRIBE_INSTANCES_RESULT | jq -r '.CpuCount' )
-if [ "$CPU_COUNT" -gt 30 ] ; then
-    echo $(min_number $MAX_POD_CEILING_FOR_HIGH_CPU $max_pods)
+if [ -n "$MAX_PODS_CEILING" ] ; then
```
Comment (suket22) [inline files/max-pods-calculator.sh:150]: Apologies for not looking at this PR earlier.

Can we use a combination of [this PR](https://github.com/awslabs/amazon-eks-ami/pull/766/files) to achieve this? You can do the `min_number` logic on whatever invokes this script ?

**Comment by o6uoq** [comment]
+1 for having a ceiling but also the option to, within the script, to provide an override.

File: files/max-pods-calculator.sh (hunk 1 above)
Comment (a7i) [reply files/max-pods-calculator.sh:150]: No worries @suket22 

I'm not sure what `SHOW_MAX_ALLOWED` would do here

**Comment by a7i** [comment]
@cartermckinnon would you please provide a reason why this was closed and why it's not a good change?

**Comment by cartermckinnon** [comment]
I'm just going through and closing stale PR's, this one needed a rebase. I'm not really against the change if you're still interested, but there's been no other engagement and I assume you've solved this on your end by now?

**Pull request closed without merging**